x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -88,6 +91,48 @@ export const useInitSourcerer = (
   } = useDeepEqualSelector((state) => scopeIdSelector(state, SourcererScopeName.timeline));
   const { indexFieldsSearch } = useDataView();
 
+  useEffect(() => {
+    const { state: sourcererInitialState, deregister } = registerUrlParam<SourcererUrlState>({
+      urlParamKey: CONSTANTS.sourcerer,
+    });
+
+    // Initialize the store with value from UrlParam.
+    if (sourcererInitialState != null) {
+      const activeScopes: SourcererScopeName[] = Object.keys(sourcererInitialState).filter(
+        (key) => !(key === SourcererScopeName.default && scopeId === SourcererScopeName.detections)
+      ) as SourcererScopeName[];
+      activeScopes.forEach((scope) =>
+        dispatch(
+          sourcererActions.setSelectedDataView({
+            id: scope,
+            selectedDataViewId: sourcererInitialState[scope]?.id ?? null,
+            selectedPatterns: sourcererInitialState[scope]?.selectedPatterns ?? [],
+          })
+        )
+      );
```

Comment:
Nit: you could get rid of the filter and just loop through once. Up to you
```suggestion
      (Object.keys(sourcererInitialState) as SourcererScopeName[]).forEach((scope) => {
        if (!(scope === SourcererScopeName.default && scopeId === SourcererScopeName.detections)) {
          dispatch(
            sourcererActions.setSelectedDataView({
              id: scope,
              selectedDataViewId: sourcererInitialState[scope]?.id ?? null,
              selectedPatterns: sourcererInitialState[scope]?.selectedPatterns ?? [],
            })
          );
        }
      });
```

Reply:
Your suggestion looks better. I am taking it.